app/src/main/java/fr/neamar/kiss/MainActivity.java

```diff
@@ -377,6 +379,21 @@ public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
             searchEditText.requestFocus();
         }
 
+        // Pasting shared text from Sharesheet via intent-filter into kiss search bar
+        Intent receivedIntent = getIntent();
+        String receivedIntentAction = receivedIntent.getAction();
+        String receivedIntentType = receivedIntent.getType();
+        if (Intent.ACTION_SEND.equals(action) && receivedIntentType != null && "text/plain".equals(receivedIntentType)) {
+            hideKeyboard();
+            String sharedText = receivedIntent.getStringExtra(Intent.EXTRA_TEXT);
+            // making sure the shared text is not an empty string
+            if (sharedText != null && sharedText.trim().length() > 0) {
+                searchEditText.setText(sharedText);
+            } else {
+                Toast.makeText(this, R.string.shared_text_empty, Toast.LENGTH_SHORT).show();
+            }
+        }
+
```

Comment:
Wrong place for handling intent as activiy is not destroyed very often.
Can you please check if this belongs to `onResume` or `onNewIntent` method?